tests/py3/test_lookup.py

```diff
@@ -299,3 +299,13 @@ def inner(a: UnknownType):
         pass
     with pytest.raises(InvalidArgument):
         inner()
+
+
+@pytest.mark.skipif(not hasattr(typing, 'NewType'), reason='test for NewType')
+def test_resolves_NewType():
+    for t in [
+        typing.NewType('T', int),
+        typing.NewType('UnionT', typing.Optional[int]),
+        typing.NewType('NestedT', typing.NewType('T', int)),
+    ]:
+        from_type(t).example()
```

Comment:
It would be nice to add an assertion that this produces a value of the right type.